docs/source/llms/dspy/notebooks/dspy_quickstart.ipynb

```diff
@@ -0,0 +1,1265 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "application/vnd.databricks.v1+cell": {
+     "cellMetadata": {
+      "byteLimit": 2048000,
+      "rowLimit": 10000
+     },
+     "inputWidgets": {},
+     "nuid": "fa602d14-6372-4e0a-92d8-6f7db048c3b8",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "source": [
+    "[DSPy](https://dspy-docs.vercel.app/) simplifies building language model (LM) pipelines by replacing manual prompt engineering with structured \"text transformation graphs.\" These graphs use flexible, learning modules that automate and optimize LM tasks like reasoning, retrieval, and answering complex questions. \n",
+    "\n",
+    "### How does it work?\n",
+    "At a high level, DSPy optimizes prompts, selects the best language model, and can even fine-tune the model using training data.\n",
+    "\n",
+    "The process follows these three steps, common to most DSPy [optimizers](https://dspy-docs.vercel.app/docs/building-blocks/optimizers):\n",
+    "\n",
+    "1. **Candidate Generation**: DSPy finds all `Predict` modules in the program and generates variations of instructions and demonstrations (e.g., examples for prompts). This step creates a set of possible candidates for the next stage.\n",
+    "2. **Parameter Optimization**: DSPy then uses methods like random search, TPE, or Optuna to select the best configurations. Fine-tuning models can also be done at this stage.\n",
```

Comment:
best configurations => best candidate for consistency with step 1. 

Finetuning is an extra step that doesn't fit here, we can just remove this sentence to avoid confusion.

Reply:
Good call on wording!

Regarding fine tuning, isn't this a [fine tuning optimizer](https://dspy-docs.vercel.app/api/optimizers/BootstrapFinetune)? I know we chatted about it not working due to bugs, but the package does in theory support this.